doc/GettingStarted.md

```diff
@@ -0,0 +1,225 @@
+Also the generator automatically detects the CPU flags set on your system, 
+so you don't have to specify each vector extension flag yourself.
+
+### **What is the target plattform?**
+
+Tested on Linux (gcc , clang) & WSL
+
+# Setup
+
+
+### 1 - Getting the repository
+
+Clone the git repositiory of the [TSL Generator](https://github.com/db-tu-dresden/TVLGen) (or one of the forks) using the recursive flag.
+
+~~~bash
+git clone --recurse-submodules https://github.com/db-tu-dresden/TVLGen <clone directory>
+~~~
+
+ ### 2 - Install dependencies
+
+ For this project you need to have cmake (Version 3.13 or above) and python (Version 3.10.* or above) installed.
+
+ There are some python dependencies, which are needed to run the generator.
+ Install them using pip:
+
+~~~bash
+pip3 install -r requirements.txt
+~~~
+
+The package pygraphviz also depends on graphviz.
+So make sure to install graphviz as well.
+
+<!-- Todo: requirements installieren, vorab graphviz-dev installieren (sudo apt isntall graphviz-dev) -->
+
+### 3 - Run the generator
+
+As already mentioned the generator will create the TSL library.
+By default, the output is directed into the generator dir, but you can specifiy an output path like following:
+
+~~~bash
+cd <clone directory>
+cmake . -D GENERATOR_OUTPUT_PATH=<path to output directory> -B build/
+~~~
+
+The CMake system is used to detect system specific hardware properties (i.e. which cpu flags are available) and hand them over to the generator.
+This way the generated TSL library is tailored for your system.
+But you can also specify the cpu flags manually, if you want to.
+For more information on how to do that, take a look at the [Generator Usage](GeneratorUsage.md) page.
+
+<!-- Todo : switches des Generators, hilfe ausgeben, Generator direkt ausführen (was macht cmake ) -->
+
+### 4 - Include into your project
+
+Add the path to the generated TSL directory to your include path and include the TSL header (tvlintrin.h).
+In a cmake project this can be done like following:
+
+~~~cmake
+add_subdirectory(<path to TSL>)
+target_link_libraries(<target> PUBLIC tvl)
+target_include_directories(<target> PUBLIC <path to TSL>)
+~~~
+
+Then include the following header in your source file:
+
+~~~cpp
+#include <tvlintrin.h>
+~~~
+
+> **Note:** Don't include other files from the TSL directory, as they depend on the include order of tvlintrin.h.
+
+# TSL
+
+https://github.com/db-tu-dresden/TVLPrimitiveData/tree/documentation
+
+Basically TSL is a collection of elementary operations, e.g. addition, multiplication, comparisons, etc., 
+but also load and store operations.
+Those operations are called **primitives**.
+Each primitive is then mapped to hardware specific instructions using a so called **Processing Style**.
+
+Take a look at the following example:
+
+~~~c++
+template<typename ps>
+ps::base_type * add_array(ps::base_type * lhs, ps::base_type * rhs, size_t size){
```

Comment:
Don't we need a `typename` before ps::base_type?

Reply:
We do. Though I like simplicity, I would also argue to have a complete MWE, including necessary includes, etc. to have an out-of-the-box compilable example.